src/Sentry.OpenTelemetry/SentrySpanProcessor.cs

```diff
@@ -211,6 +213,21 @@ public override void OnEnd(Activity data)
         PruneFilteredSpans();
     }
 
+    private Hub? GetRealHub()
+    {
+        if (_hub is Hub hub)
+        {
+            return hub;
+        }
+
+        if (_hub is HubAdapter && SentrySdk.CurrentHub is Hub sdkHub)
+        {
+            return sdkHub;
+        }
```

Comment:
Maybe this could be done in the constructor instead of each time?

Reply:
Good call. I've refactored it into a lazy lambda as well.